lib/cppcheck.cpp

```diff
@@ -430,8 +430,11 @@ unsigned int CppCheck::checkClang(const FileWithDetails &file)
         mErrorLogger.reportOut(std::string("Checking ") + file.spath() + " ...", Color::FgGreen);
 
     // TODO: get language from FileWithDetails object
-    // TODO: this ignores the configured language
-    const bool isCpp = Path::identify(file.spath(), mSettings.cppHeaderProbe) == Standards::Language::CPP;
+    bool isCpp;
```

Comment:
`isCpp` could be initialized with `(mSettings.enforcedLang == Standards::CPP);`

Reply:
That code will go away when the language is moved into `FileWithDetails`.